src/core/unit.h

```diff
@@ -123,7 +123,7 @@ struct UnitRef {
          * that we can merge two units if necessary and correct all
          * references to them */
 
-        Unit* unit;
+        Unit *source, *target;
         LIST_FIELDS(UnitRef, refs);
```

Comment:
hmm, if this has two fields now, I think it might make sense to rename "refs" too, and call it "refs_by_target" or so, to clarify that this chains up refs by the target?